### ISSUE TITLE
Ability to provide a premade screenshot/image when using renpy.save() bypassing renpy.take_screenshot()

### ISSUE THREAD

**Issue by bober-gith** [renpy/renpy#5103]
This came to mind because generating a screenshot creates an expected spike lag. 
This freeze lasts 1-2 seconds and is not a problem/feels right to the player inside the save menu itself, but it becomes a disruption when it's for a checkpoint/autosave that happens during the game, making me add a two seconds pause when a checkpoint is reached to reduce the impression that the program is becoming irresponsive. 
In a certain amount of projects you can know at which point the checkpoint/autosave will occur if you are managing it yourself and thus could provide an accurate screenshot for it, reducing the lag spike to the bare minimum. This may have also other uses, probably.

**Comment by renpytom** [comment]
Generating a screenshot should not take 1-2 seconds. (It generally takes less that one frame.) Does this happen in other games, or is it something specific to your game?

**Comment by bober-gith** [comment]
Is there a way to test produce a save without generating a screenshot, I assumed this is what took the most resources because if I press the screenshot key it lags about the same but if this takes less than one frame then I don't know.
It must be niche-specific to games that are pretty heavy for sure, I don't think the game I'm working on is literally the only one that can suffer from this since I know games that are still significantly heavier than mine. The spike lasts about 1 sec for me and it's on a very beefy computer.

**Comment by renpytom** [comment]
1 MB save files seem quite big. I'd take a look at the save dump file to figure out what's leaking into the save file that shouldn't be. That is very likely your problem. 

Overriding the save image seems like something FileJson should be used for - take the correct screenshot, and then override what's displayed, rather than baking an incorrect image into the save itself.

**Comment by renpytom** [comment]
I'd suggest looking at save_dump, to figure out why the rollbacks are so big. They really shouldn't be - the most likely explanation is you're saving data you didn't expect to save.

**Comment by bober-gith** [comment]
It has to be that, I think:
![image](https://github.com/renpy/renpy/assets/78823063/4b490e3d-e519-4133-8e2c-5cbbe1dfcdd8)
I was looking for a way to check what is inside the rollback log, I searched for it on the doc and google but couldn't find it in the past, I made some math assuming that one line of code in between one interaction (dialogue) is what gets stored for the rollback. 
Since in some parts it averaged 10 statements between interactions, I did hard_rollback_limit x 10 = rollback_length. This was to make sure that the notify alert for the rollback limit reached would work at all times. However, it could be totally wrong calculation which could be further reduced...
